### PR TITLE
migration of GET /infra/schema endpoint

### LINKED ISSUE #4275: editoast: migrate GET /infra/schema
# Description

Migrate GET /infra/schema endpoint to editoast. That endpoint is used to provide the infra objects JSON schemas to the infra editor which then generates the correct forms.

Since there's no obvious pydantic equivalent for Rust, we'll manually generate the schema to serve and copy it in editoast. We'll add a CI job to ensure that it doesn't gets out of sync with `osrd_schemas`.

# Goal

Drop API service.

# Acceptance Criteria

- [ ] the frontend uses the editoast endpoint
- [ ] the endpoint doesn't exist anymore in API

### PR DESCRIPTION
**Pull request by leovalais**
fixes #4275

### CODE REVIEW

**Review by multun — APPROVED** [review]

File: .github/workflows/editoast.yml [hunk 1]
```diff
@@ -124,3 +124,25 @@ jobs:
...
+        - uses: actions/setup-python@v4
+          with:
+            python-version: "3.11"
+            cache: "poetry"
+        - name: Check infra_schema.json sync
+          run: |
+            cd python/osrd_schemas
+            poetry install --no-interaction --no-root
+            poetry run python osrd_schemas/infra.py > current_infra_schema.json
+            diff current_infra_schema.json ../../editoast/src/views/infra/infra_schema.json
```
Comment (flomonster) [inline .github/workflows/editoast.yml]: We should do this in `osrd_schema.yml`. So we check that only when the python schema changes.

**Review by flomonster — CHANGES_REQUESTED** [review]
We should delete this endpoint and move the json schema to the front sources.

**Review by flomonster — APPROVED** [review]
LGTM

**Review by Math-R — APPROVED** [review]
LGTM !

**Pull request merged**